docs/data/material/migration/migrating-to-v6/migrating-to-pigment-css.md

```diff
@@ -0,0 +1,542 @@
+
+<codeblock>
+
+```js before
+const FlashCode = styled('div')(
+  ({ theme, startLine = 0, endLine = startLine, lineHeight = '0.75rem' }) => ({
+    top: `calc(${lineHeight} * 1.5 * ${startLine})`,
+    height: `calc(${lineHeight} * 1.5 * ${endLine - startLine + 1})`,
+    ...theme.typography.caption,
+  }),
+);
+
+export default FlashCode;
+```
+
+```js after
+const FlashCodeRoot = styled('div')(({ theme }) => ({
+  top: `calc(var(--Flashcode-lineHeight) * 1.5 * var(--Flashcode-startLine))`,
+  height: `calc(var(--Flashcode-lineHeight) * 1.5 * (var(--Flashcode-endLine) - var(--Flashcode-startLine) + 1))`,
+  ...theme.typography.caption,
+}));
+
+const FlashCode = React.forwardRef(function FlashCode(props, ref) {
+  const {
+    children,
+    startLine = 0,
+    endLine = startLine,
+    lineHeight = '0.75rem',
+    ...other
+  } = props;
+
+  return (
+    <FlashCodeRoot
+      ref={ref}
+      {...other}
+      style={{
+        '--Flashcode-lineHeight': lineHeight,
+        '--Flashcode-startLine': startLine,
+        '--Flashcode-endLine': endLine,
+        ...other.style,
+      }}
+    >
+      {children}
+    </FlashCodeRoot>
+  );
+});
+
+export default FlashCode;
+```
+
+</codeblock>
+
+## Migrating layout components
+
+To use layout components that are compatible with Pigment CSS, replace the following components with those from the adapter package:
+
+```diff
+- import Container from '@mui/material/Container';
++ import Container from '@mui/material-pigment-css/Container';
+
+- import Grid from '@mui/material/Grid';
++ import Grid from '@mui/material-pigment-css/Grid';
+
+- import Stack from '@mui/material/Stack';
++ import Stack from '@mui/material-pigment-css/Stack';
+
+- import Hidden from '@mui/material/Hidden';
++ import Hidden from '@mui/material-pigment-css/Hidden';
+```
+
+## Migrating Box component
+
+Choose one of the following approaches:
+
+### Continue using Box component
+
+Replace the `Box` component with the one from the adapter package:
+
+```diff
+- import Box from '@mui/material/Box';
++ import Box from '@mui/material-pigment-css/Box';
+```
+
+### Replace Box with HTML element
+
+Pigment CSS can extract the `sx` prop from any JSX element, so there is no need to use the Box component.
+
+```diff
+- import Box from '@mui/material/Box';
+
+ function CustomCard() {
+   return (
+-    <Box sx={{ display: 'flex' }}>
+-      <Box component="img" src="..." sx={{ width: 24, height: 24 }}>
+-      ...
+-    </Box>
++    <div style={{ display: 'flex' }}>
++      <img src="..." style={{ width: 24, height: 24 }}>
++      ...
++    </div>
+   )
+ }
+```
+
+For **TypeScript** users, you need to extend the `HTMLAttributes` interface to support the `sx` prop. Add the following code to a file that is included in your `tsconfig.json`:
+
+```ts
+import type { Theme, SxProps } from '@mui/material/styles';
+
+declare global {
+  namespace React {
+    interface HTMLAttributes<T> {
+      sx?: SxProps<Theme>;
+    }
+  }
+}
+```
```

Comment:
We are missing the Grid migration guide.

Reply:
I am not sure if it's related to Pigment migration?

Reply:
My thought was, if you are migrating to Pigment CSS, you need to use the Pigment CSS grid. We can add it in a follow up PR, let's not block the release for it.